src/Quotations.ts

```diff
@@ -105,13 +105,48 @@ export function extractFromHtml(messageBody: string): ExtractFromHtmlResult {
   if (numberOfCheckpoints >= NodeLimit)
     return { body: messageBody, didFindQuote: false, isTooLong: true };
 
-  const quotationCheckpoints = new Array<boolean>(numberOfCheckpoints);
-  const messagePlainText = preprocess(elementToText(xmlDocument), "\n", ContentTypeTextPlain);
+  let extractQuoteHtml = extractQuoteHtmlViaMarkers(numberOfCheckpoints, xmlDocument, false);
+  if (extractQuoteHtml.error)
+    return { body: messageBody, didFindQuote: false, isTooLong: true };
+
+  // Make sure we did not miss a quote due to some parsing error
+  if (!extractQuoteHtml.quoteWasFound)
+    extractQuoteHtml = extractQuoteHtmlViaMarkers(numberOfCheckpoints, xmlDocument, true);
+
+  if(extractQuoteHtml.quoteWasFound) {
+    // Remove the tags that we marked as quotation from the HTML.
+    deleteQuotationTags(xmlDocument, xmlDocumentCopy, extractQuoteHtml.quotationCheckpoints);
+
+    // Fix quirk in XmlDom.
+    if (xmlDocumentCopy.nodeType === 9 && !xmlDocumentCopy.documentElement)
```

Comment:
I guess `9` means something. Can we add a comment or introduce a constant?

Reply:
😬 https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType

Reply:
But yeah we could make an ENUM.